site/bps/BP-35-128-bits-support.md

```diff
@@ -0,0 +1,399 @@
+---
+title: "BP-34: 128 bits support"
+issue: https://github.com/apache/bookkeeper/603
+state: "Under Discussion"
+design doc: https://docs.google.com/document/d/1cu54dNSV2ZrdWCi40LcyX8NxXGRCW0609T_ewmK9BWM
+release: "4.9.0"
+---
+
+### Motivation
+
+BookKeeper coordinates with a metadata store to generate a cluster wide `ledgerId`.
+Currently this is a signed `64 bit` number (effectively 63 bits). This method works
+great because we have a centralized metadata store for coordinating the id generation.
+However this method may not scale as the cluster size and number of ledgers grow.
+
+[Universally unique identifier - Wikipedia](https://en.wikipedia.org/wiki/Globally_unique_identifier)
```

Comment:
That's my understanding too. So in that way, should we just keep calling it lid_usb? or something? naming it ```scope``` might make folks see things differently (mostly in the wrong way)

Reply:
`UUID` is one way for generating 128 bit. However it might be other ways to generate 128 bit ledger id. The angle I am coming from is adding another `64 bit` to existing `64 bit`, so that's why I call it `scopeId`. In that way, if people use UUID for ledger id generation, it can make msb to scopeId and lsb to existing ledger id; and it will also give the flexibility for people to decide how they want to generate the id and how they want to map it into scopeId and ledgerId.

call it "lid_msb" is fine to me.

Reply:
I'm fine with it being scopeID. I don't think we should tie ourselves to any one ID generation mechanism. I can think of many ways a scope ID mechanism could be used in a decentrallized manner without using UUID. Each client could even be given a scopeId on boot, given that there's 64bits to play with.

Reply:
Great